php/WcaOauth.php

```diff
@@ -19,12 +21,27 @@ class WcaOauth
 
     public function __construct($options)
     {
+        $this->setOptions($options);
+        $this->checkRequiredOptionsSet();
+    }
+
+    protected function checkRequiredOptionsSet()
+    {
         foreach (self::$requiredOptions AS $value) {
-            if (!isset($options[$value])) {
+            if (!isset($this->$value)) {
                 throw new Exception("$value is a required option!");
             }
+        }
+    }
+
+    protected function setOptions($options)
+    {
+        foreach ($options AS $key => $value) {
+            if (is_array($value)) {
+                $value = implode(' ', $value);
```

Comment:
@jfly, how do you feel about this? I debated this internally for a bit. The purpose of this is the scope parameter as it ends up being a "space separated list". This definitely needs imploded at **some point** so that the url encoding correctly interprets this is to be a space separated list. I wasn't sure if this burden should be on the oauth client, or on the oauth class.

Here is why the implode needs to happen:
With implode:
https://www.worldcubeassociation.org/oauth/authorize?client_id=064a2faa0c41616d60e0c3816d02e3d9c2cd135233517fba923fcf58f05bcede&redirect_uri=http://localhost:8080/oauth-client.php&response_type=code&scope[0]=public&scope[1]=dob&scope[2]=email

Without Implode:
https://www.worldcubeassociation.org/oauth/authorize?client_id=064a2faa0c41616d60e0c3816d02e3d9c2cd135233517fba923fcf58f05bcede&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Foauth-client.php&response_type=code&scope=public+dob+email

You can see how url encoding an array literally creates an array in the URI. This is not the behavior we want.

I'm making an assumption here that WCA OAuth will never have array syntax in a URI like that. That is quite an assumption, and can definitely break how the oauth client test script works.

Just for reference, here is how I'm currently using the oauth class:
```php
$wca = new WcaOauth(array(
    'applicationId' => '064a2faa0c41616d60e0c3816d02e3d9c2cd135233517fba923fcf58f05bcede',
    'applicationSecret' => '2b336fbce373bced434812c2aff4a2003af084eb716fc5426ec2edff3ff0da68',
    'redirectUri' => 'http://localhost:8080/oauth-client.php',
    'scope' => ['public', 'dob', 'email'],
));
```

That keeps the code nice and clean. However, if in the future the WCA decides to have arrays in the URI for whatever reason, we would have to get rid of this implode line, and move it to here:
```php
$wca = new WcaOauth(array(
    'applicationId' => '064a2faa0c41616d60e0c3816d02e3d9c2cd135233517fba923fcf58f05bcede',
    'applicationSecret' => '2b336fbce373bced434812c2aff4a2003af084eb716fc5426ec2edff3ff0da68',
    'redirectUri' => 'http://localhost:8080/oauth-client.php',
    'scope' => implode(['public', 'dob', 'email']),
));
```

Thoughts/comments?